### PR TITLE
feat: logging: add context logger helpers

### PR DESCRIPTION
**Pull request by svfoxat**
## Description

https://dt-rnd.atlassian.net/browse/ICP-2916

- adds three helpers to `pkg/logd/context.go` to thread the controller-runtime–seeded logger through `context.Context`:
  - `NewFromContext(ctx, name, keysAndValues...)` — call at package entry points; appends a name and stores the enriched logger back into ctx
  - `FromContext(ctx)` — call in downstream helpers within the same package; falls back to `logd.Get()` when no logger is in ctx
  - `IntoContext(ctx, log)` — explicitly seed or replace a logger in ctx
- updates `doc/coding-style-guide.md` with the new pattern: `NewFromContext` at package boundaries, `FromContext` inside, `var log` globals only in `cmd/`

### Stack
- **PR 1 (this):** https://github.com/Dynatrace/dynatrace-operator/pull/6472
- PR 2: https://github.com/Dynatrace/dynatrace-operator/pull/6473
- PR 3: https://github.com/Dynatrace/dynatrace-operator/pull/6474

## How can this be tested?
`make go/test`

### CODE REVIEW

File: doc/coding-style-guide.md [hunk 1]
```diff
@@ -463,15 +463,39 @@ import (
...
+  }
+  ```
+
+- **Use `logd.IntoContext(ctx, log)` when explicitly seeding a logger** into a context (e.g. in tests or when constructing a sub-context manually):
+
+  ```go
+  ctx = logd.IntoContext(ctx, myLogger)
+  ```
+
+- **Utility functions that emit logs should accept `log logd.Logger` as a parameter** rather than closing over a package-level `log`.
```
Comment (avorima) [inline doc/coding-style-guide.md]: They should accept a context

File: doc/coding-style-guide.md [hunk 2]
```diff
@@ -489,6 +513,17 @@ import (
...
 
+### Exceptions
+
+`cmd/` packages and functions that have no `context.Context` parameter (e.g. background singletons, startup helpers) may keep a package-level logger declared with `logd.Get().WithName(...)`. Document the exception with a comment:
+
+```go
+// log is kept as a package-level logger because GetMinorVersion is a background singleton
+// cache with no context.Context parameter. This is an intentional exception to the
+// context-logger pattern.
+var log = logd.Get().WithName("k8sversion")
```
Comment (avorima) [inline doc/coding-style-guide.md]: I had actually planned to plumb the context through to all of these "singleton" loggers. There really isn't any reason to use a package variable anymore

File: pkg/logd/context.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	return logr.NewContext(ctx, log.Logger), log
+}
+
+func FromContext(ctx context.Context) Logger {
+	if ctx == nil {
+		return Get()
+	}
+
+	logger, err := logr.FromContext(ctx)
+	if logger.GetSink() == nil || err != nil {
```
Comment (avorima) [inline pkg/logd/context.go:21]: the discard logger has a nil sink, so this function would prevent the use of it. if you check [controller-runtime's implementation](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.23.3/pkg/log/log.go#L91) they don't have this behavior

**Comment by avorima** [comment]
<strike>Hold on, this doesn't do want we want yet. We're still using our dumb logger instead of the enriched logger from the controller-runtime</strike>

Nvm, the propagation works

File: doc/coding-style-guide.md (hunk 2 above)
Comment (avorima) [reply doc/coding-style-guide.md]: Looking at the few exceptions most of them are just because there is no package logger in the commands, which could easily be fixed. I guess a good rule of thumb whether a package should have a logger or not is whether it's an entrypoint, i.e. main

File: pkg/logd/context_test.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+import (
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/logd"
+	"github.com/go-logr/logr"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFromContext(t *testing.T) {
```
Comment (mihaitanasedt) [inline pkg/logd/context_test.go:11]: We can also test the case when ctx is nil

**Review by avorima — APPROVED** [review]
